DuckDuckGo/Bookmarks/Model/BookmarkList.swift

```diff
@@ -44,7 +44,14 @@ struct BookmarkList {
 
     private(set) var allBookmarkURLsOrdered: [IdentifiableBookmark]
     private var favoriteBookmarksOrdered: [IdentifiableBookmark]
-    private var itemsDict: [String: [Bookmark]]
+    private var itemsDict: [String: [Bookmark]] {
+        didSet {
+            updateLowercasedItemsDict()
+        }
+    }
+
+    // Copy of itemsDict used for efficient lookups of variant URLs
+    private(set) var lowercasedItemsDict: [String: [Bookmark]] = [:]
```

Comment:
Does it have to be `[String: [Bookmark]]` if we're only ever use `lowercasedItemsDict[key].first`?

Reply:
I wanted to keep things aligned in case we want to use the dictionary in future for something else, but I can refactor it.

Reply:
Got it. I think that given it's not persisted anywhere, we don't have to think about future-proofing here, and what we're interested in at the moment is a look-up table for any bookmark matching a given lowercase URL. It should be easy to extend into `[String: [Bookmark]]` later on, if necessary.

Reply:
Now I see that in `itemsDict` we also only ever use the `.first` bookmark. I guess we can keep it as is, and take a closer look at BookmarkList at a later time.